src/index.js

```diff
@@ -326,6 +308,8 @@ class CalendarHeatmap extends React.Component {
   }
 
   render() {
+    this.valueCache = this.getValueCache(this.props.values);
```

Comment:
Can you please move this to `componentDidUpdate`, or otherwise out of `render`? It’s an antipattern in React to tie rendering to logic in any way. (creating an arbitrary `this.valueCache` isn't quite as dicey as using `this.setState`, which I think has its own warning, but it's still the kind of side effect that'll cause problems down the line.)

Reply:
I replied my reasoning of making it in the render() to review by evan-007 above.